actors/builtin/miner/miner_state.go

```diff
@@ -965,25 +974,24 @@ func (st *State) AddLockedFunds(store adt.Store, currEpoch abi.ChainEpoch, vesti
 		vestThisTime := big.Sub(targetVest, vestedSoFar)
 		vestedSoFar = targetVest
 
-		// Load existing entry, else set a new one
-		key := EpochKey(vestEpoch)
-		lockedFundEntry := big.Zero()
-		_, err = vestingFunds.Get(key, &lockedFundEntry)
-		if err != nil {
-			return err
-		}
-
-		lockedFundEntry = big.Add(lockedFundEntry, vestThisTime)
-		err = vestingFunds.Set(key, &lockedFundEntry)
-		if err != nil {
-			return err
+		// epoch already exists. Load existing entry
+		// and update amount.
+		if index, ok := epochToIndex[vestEpoch]; ok {
+			currentAmt := st.VestingFunds[index].Amount
+			st.VestingFunds[index].Amount = big.Add(currentAmt, vestThisTime)
+		} else {
+			// create a new entry and insert at the correct position
+			// st.VestingFunds is sorted by epoch
+			entry := VestingFund{Epoch: vestEpoch, Amount: vestThisTime}
+			st.VestingFunds = append(st.VestingFunds, entry)
 		}
 	}
 
-	st.VestingFunds, err = vestingFunds.Root()
-	if err != nil {
-		return err
-	}
+	// sort slice by epoch
+	sort.Slice(st.VestingFunds, func(first, second int) bool {
+		return st.VestingFunds[first].Epoch <= st.VestingFunds[second].Epoch
```

Comment:
```suggestion
		return st.VestingFunds[first].Epoch < st.VestingFunds[second].Epoch
```

(shouldn't matter anyways)

Reply:
This is done.